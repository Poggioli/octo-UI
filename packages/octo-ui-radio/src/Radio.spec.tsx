import { render } from "@testing-library/react";
import {
  RadioGroup,
  RadioGroupIndicator,
  RadioGroupItem,
  RadioGroupItemContainer,
  RadioGroupItemLabel,
} from "./Radio";

describe("<Radio />", () => {
  const mount = () =>
    render(
      <RadioGroup defaultValue="comfortable" aria-label="View density">
        <RadioGroupItemContainer>
          <RadioGroupItem disabled value="default" id="r1" />
          <RadioGroupItemLabel htmlFor="r1">Default</RadioGroupItemLabel>
        </RadioGroupItemContainer>
        <RadioGroupItemContainer>
          <RadioGroupItem value="comfortable" id="r2" />
          <RadioGroupItemLabel htmlFor="r2">Comfortable</RadioGroupItemLabel>
        </RadioGroupItemContainer>
        <RadioGroupItemContainer>
          <RadioGroupItem value="compact" id="r3" />
          <RadioGroupItemLabel htmlFor="r3">Compact</RadioGroupItemLabel>
        </RadioGroupItemContainer>
      </RadioGroup>
    );

  it(`GIVEN a <Radio />
      WHEN render is called
      THEN should render a <Radio />`, () => {
    const rendered = mount();
    expect(rendered.container).toMatchSnapshot();
  });

  it(`GIVEN a <RadioGroup />
      WHEN toString is called
      THEN return a string`, () => {
    expect(RadioGroup.toString()).toBe(".octo-UI-c-beTnUS");
  });

  it(`GIVEN a <RadioGroupItemContainer />
      WHEN toString is called
      THEN return a string`, () => {
    expect(RadioGroupItemContainer.toString()).toBe(".octo-UI-c-gyTDmN");
  });

  it(`GIVEN a <RadioGroupItem />
      WHEN toString is called
      THEN return a string`, () => {
    expect(RadioGroupItem.toString()).toBe(".octo-UI-c-opsBT");
  });

  it(`GIVEN a <RadioGroupItemLabel />
      WHEN toString is called
      THEN return a string`, () => {
    expect(RadioGroupItemLabel.toString()).toBe(".octo-UI-c-gHWBzz");
  });

  it(`GIVEN a <RadioGroupIndicator />
      WHEN toString is called
      THEN return a string`, () => {
    expect(RadioGroupIndicator.toString()).toBe(".octo-UI-c-deeeBS");
  });
});
