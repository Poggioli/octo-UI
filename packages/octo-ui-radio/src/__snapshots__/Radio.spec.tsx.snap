// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Radio /> GIVEN a <Radio />
      WHEN render is called
      THEN should render a <Radio /> 1`] = `
<div>
  <div
    aria-label="View density"
    aria-required="false"
    class="octo-UI-c-beTnUS octo-UI-c-beTnUS-jnbsLi-color-unstyled"
    dir="ltr"
    role="radiogroup"
    style="outline: none;"
    tabindex="0"
  >
    <div
      class="octo-UI-c-gyTDmN"
    >
      <button
        aria-checked="false"
        class="octo-UI-c-opsBT"
        data-disabled=""
        data-radix-collection-item=""
        data-state="unchecked"
        disabled=""
        id="r1"
        role="radio"
        tabindex="-1"
        type="button"
        value="default"
      />
      <label
        class="octo-UI-c-gHWBzz octo-UI-c-kwAZbD octo-UI-c-gWerhV octo-UI-c-gWerhV-iMVRpC-size-3 octo-UI-c-gWerhV-iisyIHw-css"
        for="r1"
      >
        Default
      </label>
    </div>
    <div
      class="octo-UI-c-gyTDmN"
    >
      <button
        aria-checked="true"
        class="octo-UI-c-opsBT"
        data-radix-collection-item=""
        data-state="checked"
        id="r2"
        role="radio"
        tabindex="-1"
        type="button"
        value="comfortable"
      >
        <span
          class="octo-UI-c-deeeBS"
          data-state="checked"
        />
      </button>
      <label
        class="octo-UI-c-gHWBzz octo-UI-c-kwAZbD octo-UI-c-gWerhV octo-UI-c-gWerhV-iMVRpC-size-3 octo-UI-c-gWerhV-iisyIHw-css"
        for="r2"
      >
        Comfortable
      </label>
    </div>
    <div
      class="octo-UI-c-gyTDmN"
    >
      <button
        aria-checked="false"
        class="octo-UI-c-opsBT"
        data-radix-collection-item=""
        data-state="unchecked"
        id="r3"
        role="radio"
        tabindex="-1"
        type="button"
        value="compact"
      />
      <label
        class="octo-UI-c-gHWBzz octo-UI-c-kwAZbD octo-UI-c-gWerhV octo-UI-c-gWerhV-iMVRpC-size-3 octo-UI-c-gWerhV-iisyIHw-css"
        for="r3"
      >
        Compact
      </label>
    </div>
  </div>
</div>
`;
